roles/generate_manifests/README.md

```diff
@@ -1,3 +1,20 @@
-# generate_manifests
+# generate_manifests role
 
-Generates the manifests required for openshift_installer's agent sub-command
\ No newline at end of file
+Generates the manifests required for openshift_installer's agent sub-command
+
+## Variables
+
+All of the variables have sane defaults which you can override to force things, but you shouldn't need to
+
+| Variable                      | Default                           | Required  | Description           |
+| ----------------------------- | --------------------------------- | --------- | --------------------- |
+| generated_dir                 | $REPO_ROOT_PATH/generated         | Yes       | Base path for all generated files |
+| manifests_dir                 | $GENERATED_DIR/$CLUSTER_NAME      | Yes       | Path to store all rendered manifests |
+| cluster_manifest_dir          | $MANIFESTS_DIR/cluster-manifests  | Yes       | Path for cluster manifests |
+| extra_manifest_dir            | $MANIFESTS_DIR/openshift          | Yes       | Path for extra manifests |
+| static_network_config         | {}                                | No        | Static network config for every node |
+| root_device_hints             | {}                                | No        | Device Hints per node |
```

Comment:
could you include the doc for the root device hints you shared in the conversation, I think is quite useful to understand how to use that variable

https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html-single/installing_an_on-premise_cluster_with_the_agent-based_installer/index#root-device-hints_preparing-to-install-with-agent-based-installer